addons/partner_autocomplete/models/res_partner.py

```diff
@@ -130,6 +132,25 @@ def read_by_vat(self, vat, timeout=15):
         if vies_vat_data:
             return [self._format_data_company(vies_vat_data)]
         else:
+            try:
+                vies_result = check_vies(vat)
+                if vies_result and vies_result['valid'] and (name := vies_result['name']) != '---':
```

Comment:
We can't use the walrus operator in Odoo 15 as it's only available in Python 3.8+ and we must support Python 3.7 in 15.